### ISSUE TITLE
panic: apps keep restarting by themselves; apps race to get focus 

### ISSUE THREAD

**Issue by biocyberman** [koekeishiya/kwm#250]
I am implementing the suggestion over here: https://github.com/koekeishiya/kwm/issues/244
This is my launch script:

```bash
#!/bin/bash
# filename: ~/.kwm/launchapp
appname=$1
shift
workspace=$1
shift
openopts="$@"
kwmc=/usr/local/bin/kwmc
$kwmc space -s id $workspace
echo apname: $appname
echo workspace: $workspace
if [[ ! "$appname" =~ "/Applications/" ]]; then
    appname="/Applications/"${appname}
fi

appname=${appname}.app
if [ ! -d "$appname" ]; then
    echo Application not found: $appname
    echo Make sure it exists with correct name and case
    exit 1
fi
open $openopts $appname
```

And snippet from ~/.kwm/kwmrc

```kwm
kwmc config spaces-key ctrl
kwmc bind ctrl-1 sys ~/.kwm/launchapp Emacs 1 -a
kwmc bind ctrl-3 sys ~/.kwm/launchapp Firefox 3 -a
```
After running `kwmc config reload`, the shortcut keys work. But there are very disturbing symtoms:

1. Apps (i.e. Emacs and Firefox) keep crashing and restarting themselves. Various ways in attempt to close the apps fail most of the time. 
2. Can't drag apps to other Desktop after hitting `F9` to open mission control
3. Apps do not seem to open at designated Desktops (i.e. Firefox open at whichever current Desktop I am at)
4. Wait the apps for about 5 minutes and I can close them. 

## System
- Mac OS El Captitan
- Have 6 workspaces/Desktop
- First set "Assign to Desktop on Display 1" after right-clicking `iTerm`'s icon; and "Assign to Desktop on Display 2" after right-clicking Firefox's icon. When the problems occurred, I set "Assign to None".
- Two monitor
- Other applications opened by other ways (i.e. normal mouse clicking on App's icon or `CMD+RET` for `iTerm` do not suffer this problem. However, apps goes back to the Desktop before they were exited. So maybe `kwmc bind ... sys` does not interact with `launchapp` script smoothing.

**Comment by koekeishiya** [comment]
> workspace=$1

From how you call the script, shouldn't this be $2 ?
I use the following, and it works fine:
```
kwmc=/usr/local/bin/kwmc
space=$1
app=$2

$kwmc space -s id $space
open -n $app
```

kwmrc: `kwmc bind cmd+alt+ctrl-n sys ~/.kwm/scripts/launch.sh 1 /Applications/Notes.app`

**Comment by biocyberman** [comment]
@koekeishiya Thanks for looking into this. The `workspace=$1` is ok because I have the `shift` comand right after that.

**Comment by koekeishiya** [comment]
The issue is probably because of the "Assign to Desktop". I have no idea how this works.

**Comment by biocyberman** [comment]
- I do not have more time at the moment to look into this further. However, I narrowed down the problem to the collision of `kwmc config spaces-key ctrl`  and  `kwmc bind ctrl-1`.  Changing `kwmc bind ctrl-1` to something else, like `kwmc bind cmd+alt+ctrl-1` makes the app stop crashing.  

- Another trick I found that hitting `Cmd+alt+ctrl-q` to quit `kwm` and so `kwm` restart by itself stop the crashing problem temporarily. This let me quit the crashing app (i.e. Firefox). 

- Lastly, I also want to inform that: having a minimum lauch script like yours but keeping the same `kwmrc` like I had in the first post made the app only cash and restart but not blink. So somehow my original launchapp script cause the app being launched to crash more miserably with blinking and nagging (keep hijacking focus).
